content/blog/scales-1-4-0/index.Rmd

```diff
@@ -0,0 +1,182 @@
+scales is a package that provides much of the scaling logic that is used in ggplot2 to a general framework, along with utility functions for e.g. formatting labels or creating colour palettes.
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("scales")
+```
+
+This blog post will give an overview of the 1.4.0 release, which has some nifty upgrades for working with colours.
+
+You can see a full list of changes in the [release notes](https://scales.r-lib.org/news/index.html)
+
+```{r setup}
+library(scales)
+```
+
+```{r ragg, include=FALSE}
+knitr::opts_chunk$set(dev = "ragg_png")
+```
+
+## Colour manipulation
+
+The `alpha()` and `muted()` functions have been part of scales for a long time. 
+Back in the 1.1.0 release we swapped to [farver](https://farver.data-imaginist.com/) to power these functions.
+We felt it was appropriate to use this package for other common colour tasks, and so `col_shift()`, `col_lighter()`, `col_darker()`, `col_saturate()` and `col_mix()` were born.
+
+```{r}
+my_colours <- c("red", "green", "blue")
+
+m <- rbind(
+  original = my_colours,
+  shift    = col_shift(my_colours, 90),
+  lighter  = col_lighter(my_colours, 20),
+  darker   = col_darker(my_colours, 20),
+  duller   = col_saturate(my_colours, -50),
+  mixed    = col_mix(my_colours, "orchid")
+)
+
+show_col(t(m), ncol = ncol(m))
+text(x = ncol(m) + 0.25, y = -(1:nrow(m)) + 0.5, rownames(m), adj = 0)
+```
+
+## Palettes
+
+Also palettes have been reworked this release to reflect more useful properties.
+Palettes now come in one of two classes: 'pal_discrete' or 'pal_continuous'.
+
+```{r}
+my_palette <- manual_pal(c("palegreen", "deepskyblue", "magenta"))
+class(my_palette)
+```
+
+Having palettes as a class rather than plain functions, allows us to store useful metadata about the palette.
+In addition, most colour palette functions also allow the aforementioned colour manipulation functions to work on the palette output.
+
+```{r}
+palette_type(my_palette)
+
+palette_nlevels(my_palette)
+
+col_shift(my_palette, 180)(3)
+```
+
+This metadata can then be used to expand discrete palettes to continuous palettes with `as_continuous_pal()` or vise versa to chop up a continuous palette into discrete palettes with `as_discrete_pal()`.
+
+```{r}
+plot(as_continuous_pal(my_palette))
+```
+
+Another thing to make working with palettes easier, is that the 'scales' package now keeps track of named palettes.
+By default, the collection of 'known' palettes is pre-populated with colour palettes from the grDevices, RColorBrewer and viridisLite packages.
+
+
+```{r}
+head(palette_names())
+
+get_palette("Okabe-Ito")(8)
+```
+
+If you're a developer of a palette package, you can use `set_palette()` to register your palette.
+This has the advantage that your palette is now available to users by name, which at times might be more convenient than having to call the palette generator function.
```

Comment:
Maybe something about not having to provide a `scale_*()` function for each palette in the future (once ggplot2 has been updated)

Reply:
Good call